main.py

```diff
@@ -121,8 +162,69 @@ def write_candidate(perm: Permuter, source: str) -> None:
             break
         except FileExistsError:
             pass
-    print(f"wrote to {fname}")
+    print(f"wrote to {fname}")   
+    
+def post_score(context: EvalContext, permuter: Permuter, cand: Candidate, exception: BaseException, profiler: Profiler) -> None:
+    if exception != None:
+        print(f"[{permuter.unique_name}] internal permuter failure.")
+        traceback.print_exc()
+        print(f"To reproduce the failure, rerun with: --seed {permuter.seed}")
+        exit(1)
+
+    if options.print_diffs:
+        permuter.print_diff(cand)
+        yn = input("Press any key to continue...")
+
+    context.iteration += 1
+    if cand.score_value is None:
+        context.errors += 1
+    disp_score = 'inf' if cand.score_value == permuter.scorer.PENALTY_INF else cand.score_value
+    timings = ''
+    if options.show_timings:
+        assert(profiler)
+        for stattype in profiler.time_stats:
+            context.overall_profiler.add_stat(stattype, profiler.time_stats[stattype])
+        timings = context.overall_profiler.get_str_stats()
+    status_line = f"iteration {context.iteration}, {context.errors} errors, score = {disp_score}\t{timings}"
+
+    if cand.score_value and cand.score_value <= permuter.base.score_value and cand.score_hash and cand.score_hash not in permuter.hashes and cand.score_value < 20000:
```

Comment:
20000? I can agree that large scores are often spammy, but we should really use a low base score to get rid of them, and perhaps require score <= some constant * base score for large base scores (perhaps with a command-line configurable constant?)

Reply:
whoops. That was debug code!